File: public/docs/_examples/toh-5/dart/lib/dashboard_component.dart
// #docplaster
// #docregion , imports
import 'dart:async';

import 'package:angular2/core.dart';
// #docregion import-router
import 'package:angular2/router.dart';
// #enddocregion import-router

import 'hero.dart';
import 'hero_service.dart';
// #enddocregion imports

// #docregion metadata
@Component(
    selector: 'my-dashboard',
    templateUrl: 'dashboard_component.html',
    // #enddocregion metadata
    // #docregion css
    styleUrls: const ['dashboard_component.css'],
    // #enddocregion css
    // #docregion metadata
    directives: const [ROUTER_DIRECTIVES],
    )
// #enddocregion metadata
// #docregion class
class DashboardComponent implements OnInit {
  List<Hero> heroes;

  // #docregion ctor
  final HeroService _heroService;

  DashboardComponent(this._heroService);
  // #enddocregion ctor

  Future<Null> ngOnInit() async {
    heroes = (await _heroService.getHeroes()).skip(1).take(4).toList();
  }
}

File: public/docs/_examples/testing/ts/app/dashboard.component.ts
// #docplaster
// #docregion
import { Component, OnInit } from '@angular/core';
// #docregion import-router
import { Router } from '@angular/router-deprecated';
// #enddocregion import-router

import { Hero } from './hero';
import { HeroService } from './hero.service';

@Component({
  selector: 'my-dashboard',
  // #docregion template-url
  templateUrl: 'app/dashboard.component.html',
  // #enddocregion template-url
  // #docregion css
  styleUrls: ['app/dashboard.component.css']
  // #enddocregion css
})
// #docregion component
export class DashboardComponent implements OnInit {

  heroes: Hero[] = [];

// #docregion ctor
  constructor(
    private _router: Router,
    private _heroService: HeroService) {
  }
// #enddocregion ctor

  ngOnInit() {
    this._heroService.getHeroes()
      .then(heroes => this.heroes = heroes.slice(1, 5));
  }

  // #docregion goto-detail
  gotoDetail(hero: Hero) {
    let link = ['HeroDetail', { id: hero.id }];
    this._router.navigate(link);
  }
  // #enddocregion goto-detail
}
// #enddocregion
